repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/headers/HeaderPassthroughTest.groovy

```diff
@@ -184,113 +169,114 @@ class SplitHeadersOnCommaTest extends ReposeValveTest {
-        "Content-Encoding" | "identity, identity, identity"                                       | 3
-        "Content-Language" | "da"                                                                 | 1
-        "Content-Language" | "mi, en"                                                             | 2
-        "Content-Language" | "en, en-US, en-cockney, i-cherokee, x-pig-latin"                     | 5
-        "Pragma"           | "no-cache"                                                           | 1
-        "Pragma"           | "no-cache, x-pragma"                                                 | 2
-        "Pragma"           | "no-cache, x-pragma-1, x-pragma-2"                                   | 3
-        "Warning"          | "199 fred \"Misc. warning\""                                         | 1
-        "Warning"          | "199 fred \"Misc. warning\", 199 ethel \"Another warning\""          | 2
-        "WWW-Authenticate" | "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="                                 | 1
-        "WWW-Authenticate" | "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==, Basic QWxhZGRpbjpwbGVhc2U/Cg==" | 1
-        "Accept-Encoding"  | "identity"                                                           | 1
-        "Accept-Encoding"  | "*"                                                                  | 1
-        "Accept-Encoding"  | "compress, gzip"                                                     | 2
-        "Accept-Encoding"  | "gzip;q=1.0, identity;q=0.5, *;q=0"                                  | 3
-        "Accept-Encoding"  | "gzip;q=0.9, identity;q=0.5, *;q=0.1"                                | 3
-        "Accept-Encoding"  | "gzip;q=0.9, *;q=0.1, identity;q=0.5"                                | 3
-        "Accept-Encoding"  | "identity;q=0.5, gzip;q=0.9, *;q=0.1"                                | 3
-        "Accept-Encoding"  | "gzip;q=0.9, identity;q=0.5"                                         | 2
-        "Accept-Encoding"  | "gzip;q=0.9, identity;q=0.5 *;q=0.6"                                 | 2
-        "Accept-Encoding"  | ""                                                                   | 0
+        headerName         | headerValue
+        "Accept"           | "text/plain"
+        "Accept"           | "text/plain, *"
+        "Accept"           | "text/plain;q=0.2, application/xml;q=0.8"
+        "Accept-Charset"   | "ISO-8859-1"
+        "Accept-Charset"   | "ISO-8859-1, *"
+        "Accept"           | "text/plain, application/xml, application/json"
+        "Accept-Charset"   | "US_ASCII, *"
+        "Accept-Charset"   | "UTF-8, US-ASCII, ISO-8859-1"
+        "Accept-Language"  | "*"
+        "Accept-Language"  | "en-US, *"
+        "Accept-Language"  | "en-US, en-gb"
+        "Accept-Language"  | "da, en-gb;q=0.8, en;q=0.7"
+        "Accept-Language"  | "en, en-US, en-cockney, i-cherokee, x-pig-latin"
+        "Allow"            | ""
```

Comment:
This stands out to me. On responses, but not on requests, we will forward headers with no value. I suspect this is a difference in how Jetty handles messages versus how Apache HTTPClient handles messages. Of course, Jetty is still letting those empty headers out the front door.

By spec, it seems that empty headers are allowable. Here's a conversation about an empty `Accept` header:
https://stackoverflow.com/questions/12130910/how-to-interpret-empty-http-accept-header

Reply:
I would agree with that.

It actually looks like Jetty passes us the empty header, but our wrapper explicitly ignores empty values:
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala#L141

That line also leads to a discprepancy where the header name shows up in a `getHeaderNames` call, but the `getHeader` call returns null and the `getHeaders` call returns an empty enumeration.

In the response wrapper, we do not filter out empty values.

Reply:
Honestly, I'm not sure... it came in here:
https://github.com/rackerlabs/repose/commit/44c60c1e99a8d948e09305cb596ce7dc1dfbe42d#diff-95e06af2b44f5b7e6073533dd6b2cbbcR139

Which was part of:
#1845

So maybe @wdschei has more context on that. 

It seems to me that both HTTP specification and Jetty allow for empty valued headers, so we should allow for them as well.

Reply:
I don't remember the full context of what was going on there, but it centers around [this](https://github.com/rackerlabs/repose/pull/1845#pullrequestreview-80221899).
Maybe the right thing to do is to remove the `.filter(StringUtils.isNotBlank)` on those and then fix any of the broken tests.